db/migrate/20230113025043_deactivate_assignments_to_deleted_targets.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class DeactivateAssignmentsToDeletedTargets < ActiveRecord::Migration[7.0]
+  def up
+    execute <<~SQL
+      UPDATE assignments
+      SET active = false
+      FROM posts
+      WHERE posts.id = assignments.target_id
+        AND assignments.target_type = 'Post'
```

Comment:
Do we need this check? What would happen if we were to remove it? That would combine both queries into one, but does it has other side effects? 🤔

Reply:
The `assignments.target_type = 'Post'` condition isn't necessary if we're sticking with  the separate queries. 
However, removing it won't allow for the queries to be combined, we'll still need these 'join' conditions in place to  select the right assignments for update.

On the subject of combining these, I considered using CTE or subquery approach but decided to go with this in the interest of simplicity and readability.

I've update the PR to combine these via explicit joins. Do take a look and let know what you think.

Reply:
Oh right, I'm a dumbass 🤦 

Sorry, but I think the two queries were much simpler and easier to read than the one query with two `LEFT JOIN`s.

Up to you if you want to keep it or revert back to two queries.

Reply:
Yes. I agree. The two queries are more readable